### PR TITLE
Restructure Chariott repo to show the separation between intent brokering and service discovery

### LINKED ISSUE #179: [PBI] Change the directory structure to emphasize the different components of Chariott
## Description
Currently, the new service discovery system is in a separate folder from the rest of the directory structure. To finish the full separation of service discovery and intent brokering for now, restructure the repository so that these are shown as separate components.

### PR DESCRIPTION
**Pull request by ladatz**
Closes #179

## Motivation and Context
Restructure the repository to show the separation between intent brokering functionality and service discovery-only. These two operate independently from one another for now. The blueprint will use the service discovery only and this change helps simplify 

## Description

- Add the intent_brokering top-level folder and move everything that only applies to it to that folder. 
- Rename the Dockerfiles to reflect which component they build.

TODO in separate PRs: https://github.com/eclipse-chariott/chariott/issues/246 ,  https://github.com/eclipse-chariott/chariott/issues/245 , and #149

### CODE REVIEW

File: .github/workflows/notice-generation.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
       - run: |
           git config --global user.email "chariott-bot@eclipse.org"
           git config --global user.name "Automated Notice Generation Pipeline"
-          ./tools/notice_generation.sh
+          ./notice_generation.sh
```
Comment (devkelley) [inline .github/workflows/notice-generation.yaml:23]: Why did you move notice generation out of tools? I believe all of the other projects have this under a tools folder.

File: intent_brokering/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+[dev-dependencies]
+anyhow = { workspace = true }
+async-trait = { workspace = true }
+examples-common = { path = "./examples/common" }
+futures = { workspace = true }
+tokio-util = { workspace = true }
+uuid = { workspace = true }
+tokio-stream = { workspace = true }
+test-case = { workspace = true }
+
```
Comment (devkelley) [inline intent_brokering/Cargo.toml:35]: nit: remove extra lines

File: intent_brokering/examples/applications/simulated-camera/README.md [hunk 3]
```diff
@@ -8,14 +8,14 @@ manual mode where you specify the frame rate event yourself.
 ## To run the application
 
 1. Start chariott runtime by executing `cargo run` from the root directory
```
Comment (devkelley) [inline intent_brokering/examples/applications/simulated-camera/README.md:10]:
```suggestion
1. Start chariott runtime by executing `cargo run -p chariott` from the root directory
```

Or specify the intent_brokering directory

File: .github/workflows/notice-generation.yaml (hunk 1 above)
Comment (devkelley) [reply .github/workflows/notice-generation.yaml:23]: and similarly. Maybe it makes sense to have the 'buf.work.yaml' and 'build-all-containers.sh' in the tools directory as well.

**Review by devkelley — DISMISSED** [review]

File: intent_brokering/README.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
+# Service Discovery
```
Comment (ashbeitz) [inline intent_brokering/README.md:1]: Should this be titled "Intent Brokering"?

**Review by ashbeitz — COMMENTED** [review]
One comment.  Please take a look.

File: intent_brokering/README.md (hunk 4 above)
Comment (ladatz) [reply intent_brokering/README.md:1]: Good catch, thanks!

**Review by ashbeitz — APPROVED** [review]
Approved

**Review by devkelley — APPROVED** [review]
